### ISSUE TITLE
Need checkpoint after hardfork.

### ISSUE THREAD

**Issue by etkim7** [firoorg/firo#975]
I can't sync to new hardforked block

I have synced for several times but firo synced to wrong chain at last and stuck at 336890 height.
I tried to `invalidateblock` 336891's block but can't get the 336891 blockchash as firo just don't respond when I try
`firo-cli getblockhash 336890`
It just hangs on.

Why not add checkpoint in latest release? I believe there's still many nodes stuck around.
Also please do not proceed hardfork release too quickly. It's like ninja patching, many node operators can't follow up.

**Comment by justanwar** [comment]
Try `firo-cli reconsiderblock bac60b152687f66dc8e70896642b76334f04b256a6875f6c048b153aad923406 `

**Comment by etkim7** [comment]
Thanks.

**Comment by lcgogo** [comment]
./firo-cli -datadir=/data/firo reconsiderblock bac60b152687f66dc8e70896642b76334f04b256a6875f6c048b153aad923406

print nothing.

nobody@509089a7365a:/firo$ ./firo-cli -datadir=/data/firo getinfo
{
  "version": 140203,
  "protocolversion": 90030,
  "walletversion": 130000,
  "balance": 0.00992336,
  "blocks": 338761,
  "timeoffset": 0,
  "connections": 8,
  "proxy": "",
  "difficulty": 1046.377832167389,
  "testnet": false,
  "keypoololdest": 1592909664,
  "keypoolsize": 100,
  "paytxfee": 0.00000000,
  "mininput": 0.00001000,
  "relayfee": 0.00001000,
  "errors": ""
}
nobody@509089a7365a:/firo$ ./firod --version
Firo Core Daemon version v0.14.2.3-g6598f81f5
Copyright (C) 2016-2021 The Firo Core developers
Copyright (C) 2009-2021 The Bitcoin Core developers

Please contribute if you find Firo Core useful. Visit <https://firo.org/> for
further information about the software.
The source code is available from <https://github.com/firoorg/firo>.

This is experimental software.
Distributed under the MIT software license, see the accompanying file COPYING
or <https://opensource.org/licenses/MIT>

This product includes software developed by the OpenSSL Project for use in the
OpenSSL Toolkit <https://www.openssl.org> and cryptographic software written by
Eric Young and UPnP software written by Thomas Bernard.

This product includes Masternodes software developed by the Dash Core
developers <https://www.dash.org>.
nobody@509089a7365a:/firo$

**Comment by justanwar** [comment]
@lcgogo Please update to v0.14.4.0 and start with `-allowdeepreorg` flag, and open an issue if problem persists.

**Comment by etkim7** [comment]
It constantly stuck.
It syncs well with empty wallet.dat but it stuck with normal worked wallet.dat
I tried
1. sync from scratch (no blockchain, no wallet.dat) -> synced well -> changed wallet.dat to old -> reindex -> synced to wrong chain.
2. sync from scratch (no blockchain, no wallet.dat) -> synced well -> changed wallet.dat to old -> rescan -> stuck at 345959
seems like 345959 is where I tried rescan.

wallet.dat itself seems corrupted or firo sync is broken when there's wrong blockchain data in the wallet.dat

`-allowdeepreorg` doesn't help. tried reconsiderblock but the wallet just hangs.
